/armor_detector:
  ros__parameters:
  # Detector params.
    armor_type_wh_thres: 3
    max_lost_cnt: 5
    max_armors_cnt: 8
    max_v: 8
    max_delta_t: 100
    no_crop_thres: 0.01
    hero_danger_zone: 4
    color: true # red: true
                # blue: false
    no_crop_ratio: 0.002
    full_crop_ratio: 0.0001
    armor_roi_expand_ratio_width: 1.1
    armor_roi_expand_ratio_height: 1.5
    armor_conf_high_thres: 0.82

  # Spinning params.
    anti_spin_judge_high_thres: 20000.0
    anti_spin_judge_low_thres: 2000.0
    anti_spin_max_r_multiple: 4.5
    max_dead_buffer: 2
    max_delta_dist: 0.5
    min_fitting_lens: 10
    shoot_delay: 50
    max_cost: 500
    switch_max_dt: 10000.0
    
    delta_x_3d_high_thresh: 0.18
    delta_x_3d_higher_thresh: 0.25
    delta_x_3d_low_thresh: 0.10
    delta_x_3d_lower_thresh: 0.05
    delta_y_3d_high_thresh: 0.18
    delta_y_3d_higher_thresh: 0.25
    delta_y_3d_low_thresh: 0.10
    delta_y_3d_lower_thresh: 0.05

  # Camera config params.
    # camera_name: "00J90630561" 
    camera_name: "KE0200110075" 
    camera_type: 0
    camera_param_path: "src/global_user/config/camera.yaml"

  # Network model path.
    # network_path: "src/vehicle_system/autoaim/armor_detector/model/opt-0127-001.xml"
    # network_path: "src/vehicle_system/autoaim/armor_detector/model/opt-0527-002.xml"
    # network_path: "src/vehicle_system/autoaim/armor_detector/model/1106-333.xml"
    # network_path: "src/vehicle_system/autoaim/armor_detector/model/yolox.xml"
    # network_path: "src/vehicle_system/autoaim/armor_detector/model/opt-1208-001.xml"
    network_path: "src/vehicle_system/autoaim/armor_detector/model/opt-0127-001.xml"
  
  # Data saving.
    save_data: false
    save_path: "src/data/infer2_5.txt"
    save_dataset: false
  
  # Debug params.
    debug: true
    using_imu: true
    using_roi: false
    # draw_predict: true
    show_img: true
    show_fps: true
    print_letency: false
    print_target_info: false
    detect_red: true
    # disable_filter: false
    # disable_fitting: true
    # show_predict: false
    using_shared_memory: false
    show_transformed_info: false
    auto_label: false
    show_all_armors: true
    sync_transport: false

/armor_processor:
  ros__parameters:
  # Predict params.
    bullet_speed: 28.0
    max_time_delta: 1000
    max_cost: 10
    max_v: 8
    min_fitting_lens: 10
    shoot_delay: 50
    window_size: 3

  # Debug params.
    using_shared_memory: false
    disable_filter: true
    disable_fitting: false
    draw_predict: false
    show_predict: false
    show_transformed_info: false
    debug: true

  # Camera params.
    camera_type: 0 #(daheng: 0 / hik: 1 / mvs: 2 / usb: 3)
    # coord_param_name: "00J90630561" 
    coord_param_name: "KE0200110075" 
    coord_param_path: "src/global_user/config/camera.yaml"
    
  # Filter params.
    filter_param_path: "src/global_user/config/filter_param.yaml"

  # Singer model(--->CS Model).
    # (alpha:机动频率 a_max:加速度上限 p_max:最大值概率 p0:非机动概率 dt:采样时间间隔 
    #  sigma:机动加速度标准差 p:过程协方差系数 r:观测协方差系数 coeff:预测提前量系数)
    singer_model: [0.80, 5.0, 0.10, 0.80, 0.80, 0.20, 1.0, 1.0, 5.0] 
  
  # IMM model.
    #状态转移概率
    trans_prob_matrix: [0.6, 0.3, 0.05, 0.05, #row prob sum = 1
                        0.5, 0.4, 0.05, 0.05,
                        0.1, 0.1, 0.75, 0.05,
                        0.1, 0.1, 0.05, 0.75]
    #模型概率
    model_prob_vector: [0.4, 0.3, 0.15, 0.15] #prob sum = 1
    process_noise: [0.4, 0.4, 0.3, 0.3, 0.2, 0.2]
    measure_noise: [60.0, 60.0, 30.0, 30.0]

  

    
